src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -928,6 +935,13 @@ private static void EnsureInitialized()
         /// <summary>Ensures that the console has been initialized for use.</summary>
         private static void EnsureInitializedCore()
         {
+            // Initialization is only needed when input isn't redirected.
+            if (Console.IsInputRedirected)
+            {
+                s_initialized = true;
+                return;
+            }
+
```

Comment:
Is there any value in adding an assert just inside each `lock (Console.Out)` ie
```c#
lock (Console.Out)
{
     Debug.Assert(!Monitor.IsEntered(s_syncObject));
    ...
}
```

Reply:
I figured I could probably put this in the `Console.Out` getter, and when I ran tests it showed another potential deadlock:
```
Assertion failed.
     at System.Console.get_Out() in /home/tmds/repos/runtime/src/libraries/System.Console/src/System/Console.cs:line 161
     at System.ConsolePal.EnsureInitializedCore() in /home/tmds/repos/runtime/src/libraries/System.Console/src/System/ConsolePal.Unix.cs:line 945
     at System.ConsolePal.EnsureConsoleInitialized() in /home/tmds/repos/runtime/src/libraries/System.Console/src/System/ConsolePal.Unix.cs:line 931
     at System.ConsolePal.ControlCHandlerRegistrar.Register() in /home/tmds/repos/runtime/src/libraries/System.Console/src/System/ConsolePal.Unix.cs:line 1477
     at System.Console.add_CancelKeyPress(ConsoleCancelEventHandler value) in /home/tmds/repos/runtime/src/libraries/System.Console/src/System/Console.cs:line 424
```

Reply:
I've added the assert to `Console.Out` getter, and fixed the `CancelKeyPress` case also.